### PR TITLE
Navigation drawer redesign #854

### LINKED ISSUE #854: Navigation drawer redesign
**Description**:
As part of the homepage redesign https://github.com/ral-facilities/datagateway/issues/954, modify the navigation drawer to look like the mockup. The navigation drawer should open below the main app bar and should not cover the footer.

https://preview.uxpin.com/ee9832eb8102ff6013aad58becab3ea162682ce9#/pages/144657834/simulate/no-panels

**Acceptance criteria**:
- [ ] Navigation drawer looks like mock-up

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description
Navigation Drawer redesign
https://preview.uxpin.com/ee9832eb8102ff6013aad58becab3ea162682ce9#/pages/144657834/simulate/no-panels 

## Testing instructions
check page with a screen reader (footer, drawer and main appbar) 

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes #854

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
nav-drawer-2

File: src/navigationDrawer/navigationDrawer.component.tsx [hunk 1]
```diff
@@ -9,23 +9,27 @@ import {
   createStyles,
   StyleRules,
   WithStyles,
-  styled,
+  // styled,
```
Comment (joelvdavies) [inline src/navigationDrawer/navigationDrawer.component.tsx:12]: I think this should be removed.

File: src/navigationDrawer/navigationDrawer.component.tsx [hunk 2]
```diff
@@ -35,13 +39,13 @@ interface NavigationDrawerDispatchProps {
...
-  padding: '0 8px',
-  justifyContent: 'flex-end',
-}));
+// const StyledHeader = styled('div')(({ theme }) => ({
+//   ...theme.mixins.toolbar,
+//   display: 'flex',
+//   alignItems: 'center',
+//   padding: '0 8px',
+//   justifyContent: 'flex-end',
+// }));
```
Comment (joelvdavies) [inline src/navigationDrawer/navigationDrawer.component.tsx:48]: This also seems unnecessary.

File: src/navigationDrawer/navigationDrawer.component.tsx [hunk 3]
```diff
@@ -9,23 +9,27 @@ import {
   createStyles,
   StyleRules,
   WithStyles,
-  styled,
+  // styled,
 } from '@material-ui/core/styles';
-import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
+
```
Comment (joelvdavies) [inline src/navigationDrawer/navigationDrawer.component.tsx:14]: Same for this line.

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Found a few small things. Also I think you should remove `position: 'absolute'` from the STFC logo as currently it ignores all other items in the list when resized e.g. 
image
The method Louise used on the homepage is here: https://github.com/ral-facilities/datagateway/pull/985/commits/f3b537fd5c8f3584b74d67a3d00d1a2ea068d859.

**Review by joelvdavies — APPROVED** [review]
Looks good and works as expected. Reviewed changes and code coverage, CI passes.

**Pull request merged**